extras/retry_test.go

```diff
@@ -0,0 +1,363 @@
+package extras
+
+import (
+	"log"
+	"net/http"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/orian/flakyhttp"
+	posthog "github.com/posthog/posthog-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// retryTestCallback implements posthog.Callback to track success/failure
+type retryTestCallback struct {
+	t            *testing.T
+	mu           sync.Mutex
+	successCount int
+	failureCount int
+	successChan  chan posthog.APIMessage
+	failureChan  chan error
+}
```

Comment:
Each of the new test files has a pretty similar pattern of a callback struct and some methods on that struct. Would it make sense to consolidate these into a shared helper?